jax/_src/scipy/special.py

```diff
@@ -1235,6 +1235,8 @@ def sph_harm(m: Array,
     A 1D array containing the spherical harmonics at (m, n, theta, phi).
   """
 
+  lax.broadcast_shapes(jnp.shape(m), jnp.shape(n), jnp.shape(theta), jnp.shape(phi))
+
```

Comment:
This will check if the input arguments are broadcast compatible. It will raise a `ValueError` if they are not broadcast compatible.

Reply:
Hi @jakevdp Could you please check if this change is correct or not.

Reply:
I think the way to do this would be to use `jax._src.numpy.util.promote_shapes`, something like this:
```python
from jax._src.numpy.util import promote_shapes

m, n, theta, phi = promote_shapes("sph_harm", m, n, theta, phi)
```
That's assuming that the intent of this API is that all inputs are broadcast to a common shape, which is not clear to me from reading the docs, the source, or the tests.